src/containers/NoteForm/NoteForm.js

```diff
@@ -74,11 +73,15 @@ export class NoteForm extends Component {
   handleSubmit = (e) => {
     e.preventDefault()
     const { id, title } = this.state.note;
-    const { items } = this.state;
+    const newItems = this.state.items;
     const { isEdit, editNote, postNote } = this.props;
     const url = isEdit ? `http://localhost:3001/api/v1/notes/${id}` : 'http://localhost:3001/api/v1/notes'
-    isEdit ? editNote(url, { id, title, items }) : postNote(url, { id, title, items })
-    this.setState({ redirect: true })
+    
+    if (newItems[newItems.length - 1].description === '') {
+      newItems.pop()
+    }
```

Comment:
Looks like a good way to handle that last "empty" item on typing into the input.